src/Components/InfiniteScroll.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from "react";
+import { IModel, IModelData, IQueryBuilder, IQueryManager, IQueryParams, QueryBuilder } from "redux-data-service";
+import { Omit } from "redux-data-service/dist/Omit";
+import { compose, defaultProps, lifecycle, pure, setDisplayName, withProps, withState, withStateHandlers } from "recompose";
+import { debounce } from "lodash";
+
+import { Query } from "../Query";
+import { withModelQuery } from "../WithModelQuery";
+
+export interface IInfiniteScrollProps<T extends IModelData> {
+  modelName: string;
+  query: IQueryParams | IQueryBuilder;
+  containerComponent: React.ComponentType<{ onScroll: React.UIEventHandler } & any>;
+  modelComponent: React.ComponentType<{ model: T } & any>;
+  containerProps?: any;
```

Comment:
Instead of a `containerProps` prop, which will lead to developers creating inline objects that cause extra react re-renders, I wonder if we should add `[key: string]: any` to the `IInfiniteScrollProps` interface, then spread any extra props onto the `ContainerComponent` when it's rendered. What do you think?

For example:
```
<InfiniteScroll
  containerComponent={DataTable}
  modelComponent={DataTableRow}
  columns={columns}
/>
```

In the above example, the `columns` prop would be spread onto the `ContainerComponent`.

Another example:
```
<ModelField
  component={InfiniteScroll}
  containerComponent={Select}
  modelComponent={MenuItem}
  value={value}
  name="country"
/>
```
In the above example, the `ModelField` will spread the extra props onto the `InfiniteScroll` component which would spread them onto the `Select` component.

Reply:
good idea, done

Reply:
i had to put in an ugly mapProps section, would you have expected it to need that?